packages/astro/src/core/app/index.ts

```diff
@@ -300,6 +312,9 @@ export class App {
 		if (!routeData) {
 			return this.#renderError(request, { locals, status: 404 });
 		}
+		if (this.#manifest.experimentalEnv && getEnv) {
+			setGetEnv(getEnv);
+		}
```

Comment:
Is `setGetEnv` only possible to be called within `render()`? I feel like env vars should be global by default and this is only for Cloudflare? Runtimes like node and deno adapter should be able to set this once.

Reply:
I looked a little bit at the adapter api and I guess we could also expose something on `App`? So adapters like node would be able to do:
```ts
const app = new App(manifest);
app.setGetEnv((key) => {
	// whatever logic
})
```
This would call the runtime internally. Wdyt @ematipico?